payjoin/tests/integration.rs

```diff
@@ -415,14 +430,14 @@ mod integration {
                 let response = agent.post(req.url).body(req.body).send().await?;
                 let res = response.bytes().await?.to_vec();
                 payjoin_proposal.process_res(res, ctx)?;
-                // response should be 204 http
 
                 // **********************
                 // Inside the Sender:
                 // Sender checks, signs, finalizes, extracts, and broadcasts
 
                 // Replay post fallback to get the response
                 let response = agent.post(url).body(body).send().await?;
+                let (_, send_ctx) = req_ctx.extract_v2(directory.to_owned())?;
```

Comment:
You'll want to get a response with a new Request & ContextV2 here. If these lines were swapped and the request were actually used then process_res would succeed.